TODO/ibeacon-in-swift.md

```diff
@@ -166,33 +178,35 @@ func locationManager(_ manager: CLLocationManager, didRangeBeacons beacons: [CLB
+- RSSI - 捕获一个 `Int` 类型的信息用来描述设备信号强度。
 
-- Accuracy – `CLLocationAccuracy` captures a distance estimate to the beacon in meters.
+另外说明一下我们同是需要关注范围模式比起监听模式是更耗电。此外，你可能在后台使用范围模式，这是很不被认可的方案，网上的讨论中都说如果你在项目中这样使用会被苹果公司拒绝上线。
 
-- RSSI – `Int` captures the signal strength of the beacon measured in decibels.
 
-The caveat with ranging is that it consumes significantly more power than monitoring. Further, while you can technically range in the background, it is strongly frowned upon. Most online discussions warn that Apple may reject your app if you do.
+## 更多讨论
 
-## Further Discussion ##
+### 距离是否准确
 
-### Is the distance accurate? ###
+[网上很多的讨论](http://stackoverflow.com/questions/20416218/understanding-ibeacon-distancing) 关于每次范围模式事件的 `CLLocationAccuracy` 类型返回值。人们通常都将这个值解释成到设备的实际值。我的经验是你可以把这个返回值解释成实际值，但有时这个值并不精确。苹果的官方文档建议开发者使用枚举 `CLProxity` 对象的值来作为初始化方案来决定范围模式中的哪一个设备最接近，然后使用 `CLLocationAccuracy` 值与这个值进行绑定。
 
-There is a [lot of discussion online](http://stackoverflow.com/questions/20416218/understanding-ibeacon-distancing)  surrounding the `CLLocationAccuracy` value returned with each ranging event. Folks often want to interpret this as the literal distance to the beacon. In my experience, you can attempt to interpret the value as a literal distance, but it will not always be accurate. The Apple docs suggest you use the `CLProxity` enumeration value as the initial way to determine which beacons are closest, and then break ties using the `CLLocationAccuracy` value.
+[这篇博客](https://shinesolutions.com/2014/02/17/the-beacon-experiments-low-energy-bluetooth-devices-in-action/) 进行了一些有趣的试验来测试精确的距离。
 
-[This blog post](https://shinesolutions.com/2014/02/17/the-beacon-experiments-low-energy-bluetooth-devices-in-action/)  conducts some interesting experiments to test the accuracy of the distance value.
+### 监听超过20个设备
 
-### Monitoring more than 20 beacons ###
+正像我前边介绍的那样，你现在只能监听20个 iBeacons ，如果你需要监听超过20个设备，你将需要在应用运行的过程中更改监听的设置，一种实现方案是用图表来展示你的 iBeacons 网络，在网络中定义最顶层的 iBeacons 以及如果这些彼此接近的情况下能够连接到边界。这样你就可以快速的查找到最接近的20个 iBeacons 并监听他们。这需要很多的工作，但是定义一个这样的拓扑是一种实现 20个 iBeacons 限制的方式。
 
-As I mentioned earlier, you can only monitor 20 iBeacons concurrently. If you need to monitor more than 20, you will need to update the set being monitored as the app runs. One way to do this is to represent your iBeacon network as a graph, where the iBeacons are vertices and they are connected by an edge if they are near each other. When the user enters near an iBeacon, you can quickly search for its 20 nearest neighbors and monitor for them. This requires some work, but defining a topology like this is one way to work around the 20 iBeacon limitation.
+### Faraday 容器
 
-### Faraday containers ###
+如果你的 iBeacons 不能关闭或者不能轻易的在广播数据的过程中被停止，你也许应该考虑购买一个法拉第包或笼。当附近有多个设备的时候，它可以尝试着一个接一个的去配置 iBeacons。我们很确定 Estimote 的 iBeacons 就有上述的问题，放置所有的设备除了其中一个放置在法拉第笼中，这将使你很容易是你想要配置的 iBeacons 被孤立出来。（译者注：这一段确实翻译的不太好。）
```

Comment:
这个应该是如果 iBeacon 不能被关闭的话，多个设备配置起来很麻烦，会互相干扰，所以需要法拉第笼来屏蔽其他设备，一个个来设置的意思吧。

>It can be tricky attempting to configure each iBeacon, one at a time, when you have several nearby.

>当附近有多个设备的时候，尝试一次一个的配置 iBeacon 会变得相当棘手。